File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/AttributeSetInstanceExtensionFieldStateEventId.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\StringIdInterface;

class AttributeSetInstanceExtensionFieldStateEventId implements StringIdInterface
{
    /**
     * @Type("string")
     */
    private $groupId;

    /**
     * @return string
     */
    public function getGroupId()
    {
        return $this->groupId;
    }

    /**
     * @param string $groupId
     */
    public function setGroupId($groupId)
    {
        $this->groupId = $groupId;
    }

    /**
     * @Type("string")
     */
    private $index;

    /**
     * @return string
     */
    public function getIndex()
    {
        return $this->index;
    }

    /**
     * @param string $index
     */
    public function setIndex($index)
    {
        $this->index = $index;
    }

    /**
     * @Type("Dddml\Serializer\Type\Long")
     */
    private $attributeSetInstanceExtensionFieldGroupVersion;

    /**
     * @return Long
     */
    public function getAttributeSetInstanceExtensionFieldGroupVersion()
    {
        return $this->attributeSetInstanceExtensionFieldGroupVersion;
    }

    /**
     * @param Long $attributeSetInstanceExtensionFieldGroupVersion
     */
    public function setAttributeSetInstanceExtensionFieldGroupVersion($attributeSetInstanceExtensionFieldGroupVersion)
    {
        $this->attributeSetInstanceExtensionFieldGroupVersion = $attributeSetInstanceExtensionFieldGroupVersion;
    }



    /**
     * @var AttributeSetInstanceExtensionFieldStateEventIdFlattenedDto
     */
    private $idFlattenedDto;

    /**
     * @return string
     */
    public function toString()
    {
        if (!$this->idFlattenedDto) {
            $this->idFlattenedDto = new AttributeSetInstanceExtensionFieldStateEventIdFlattenedDto($this);
        }

        return $this->idFlattenedDto->toString();
    }

    /**
     * @param string $idStr
     *
     * @return AttributeSetInstanceExtensionFieldStateEventId
     */
    public static function createFromString($idStr)
    {
        return (new AttributeSetInstanceExtensionFieldStateEventIdFlattenedDto())
            ->fromString($idStr)
            ->toAttributeSetInstanceExtensionFieldStateEventId();
    }


}

File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/AttributeSetInstanceExtensionFieldStateEventId.java
package org.dddml.wms.domain;


public class AttributeSetInstanceExtensionFieldStateEventId
{
    private String groupId;

    public String getGroupId()
    {
        return this.groupId;
    }

    public void setGroupId(String groupId)
    {
        this.groupId = groupId;
    }

    private String index;

    public String getIndex()
    {
        return this.index;
    }

    public void setIndex(String index)
    {
        this.index = index;
    }

    private Long attributeSetInstanceExtensionFieldGroupVersion;

    public Long getAttributeSetInstanceExtensionFieldGroupVersion()
    {
        return this.attributeSetInstanceExtensionFieldGroupVersion;
    }

    public void setAttributeSetInstanceExtensionFieldGroupVersion(Long attributeSetInstanceExtensionFieldGroupVersion)
    {
        this.attributeSetInstanceExtensionFieldGroupVersion = attributeSetInstanceExtensionFieldGroupVersion;
    }

    AttributeSetInstanceExtensionFieldStateEventId ()
    {
    }

    public AttributeSetInstanceExtensionFieldStateEventId (String groupId, String index, Long attributeSetInstanceExtensionFieldGroupVersion)
    {
        this.groupId = groupId;
        this.index = index;
        this.attributeSetInstanceExtensionFieldGroupVersion = attributeSetInstanceExtensionFieldGroupVersion;
    }

}

